ctapipe/io/eventfilereader.py

```diff
@@ -148,216 +148,207 @@ def check_file_compatibility(file_path):
-            instead of index
+    def __iter__(self):
+        # Always reset generator when starting a new iteration
+        self.reset()
+        for event in self.source:
+            self.current_event = event
+            yield event
 
-        Returns
-        -------
-        event : `ctapipe` event-container
+    def __enter__(self):
+        return self
 
-        """
-        source = self.read(requested_event=requested_event,
-                           use_event_id=use_event_id)
-        event = next(source)
-        return deepcopy(event)
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
 
-    def find_max_true_npe(self, telescopes=None):
+    def __getitem__(self, item):
         """
-        Loop through events to find the maximum true npe
+        Obtain a particular event
 
         Parameters
         ----------
-        telescopes : list
-            List of telecopes to include. If None, then all telescopes
-            are included.
+        item : int or str
+            If `item` is an int, then this is the event_index for the event
+            obtained. If `item` is a str, then this is the event_id for the
+            event obtained.
 
         Returns
         -------
-        max_pe : int
+        event : ctapipe.io.container
+            The event container filled with the requested event's information
 
         """
-        # TODO: Find an alternate method so this can be removed
-        self.log.info("Finding maximum true npe inside file...")
-        source = self.read()
-        max_pe = 0
-        for event in source:
-            tels = list(event.dl0.tels_with_data)
-            if telescopes is not None:
-                tels = []
-                for tel in telescopes:
-                    if tel in event.dl0.tels_with_data:
-                        tels.append(tel)
-            if event.count == 0:
-                # Check events have true charge included
-                try:
-                    if np.all(event.mc.tel[tels[0]].photo_electron_image == 0):
-                        raise KeyError
-                except KeyError:
-                    self.log.exception('[chargeres] Source does not contain '
-                                       'true charge')
-                    raise
-            for telid in tels:
-                pe = event.mc.tel[telid].photo_electron_image
-                this_max = np.max(pe)
-                if this_max > max_pe:
-                    max_pe = this_max
-        self.log.info("Maximum true npe inside file = {}".format(max_pe))
-
-        return max_pe
-
-
-class HessioFileReader(EventFileReader):
-    origin = 'hessio'
+        if self.is_stream:
+            raise IOError("Input is a stream, __getitem__ is disabled")
+
+        current = None
+
+        if not self.has_fast_seek and self.getevent_warn:
+            self.log.warning("Seeking to event... (potentially long process)")
+            self.getevent_warn = False
```

Comment:
But as this is a Component it would be more appropriate to use the built-in logger, so the log messages are correctly handled according to how they are set-up. We can change this in the future if needed.

Reply:
Logging with level `warning` is something different than a true warning. Here a true warning is more appropriate.